apps/judicial-system/backend/src/app/modules/case/case.service.ts

```diff
@@ -472,6 +474,23 @@ export class CaseService {
       })
   }
 
+  private async deliverCaseToPolice(theCase: Case): Promise<boolean> {
+    const pdf = await getCourtRecordPdfAsString(theCase, this.formatMessage)
+    const defendantNationalIds = theCase.defendants
+      ?.filter((defendant) => !defendant.noNationalId && defendant.nationalId)
+      .map((defendant) => defendant.nationalId ?? '')
```

Comment:
Is the `?? ''` needed here, because you've already filtered out all defendants that don't have a value for nationalId. I know typescript is sometimes a bit weird when it comes to these kinds of checks, but I just wanted to make sure.

Reply:
Well spotted and you would think this was unnecessary. However, the type inference is not smart enough to figure this out and that is why added this redundant code.